qiskit_nature/circuit/library/initial_states/fermionic_gaussian_state.py

```diff
@@ -0,0 +1,159 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Fermionic Gaussian states."""
+
+from typing import Optional, Sequence
+
+import numpy as np
+from qiskit import QuantumCircuit, QuantumRegister
+from qiskit_nature.converters.second_quantization import QubitConverter
+from qiskit_nature.mappers.second_quantization import JordanWignerMapper
+
+from .utils.givens_rotations import _prepare_fermionic_gaussian_state_jordan_wigner
+
+
+def _validate_transformation_matrix(
+    mat: np.ndarray, rtol: float = 1e-5, atol: float = 1e-8
+) -> None:
+    if not len(mat.shape) == 2:
+        raise ValueError("transformation_matrix must be a 2-dimensional array.")
+
+    n, p = mat.shape  # pylint: disable=invalid-name
+    if p != n * 2:
+        raise ValueError("transformation_matrix must have shape (n_orbitals, 2 * n_orbitals).")
+
+    left = mat[:, :n]
+    right = mat[:, n:]
+    comm1 = left @ left.T.conj() + right @ right.T.conj()
+    comm2 = left @ right.T + right @ left.T
+    one = np.eye(n)
+    zero = np.zeros((n, n))
+    if not np.allclose(comm1, one, rtol=rtol, atol=atol) or not np.allclose(comm2, zero):
+        raise ValueError(
+            "transformation_matrix does not describe a valid transformation "
+            "of fermionic ladder operators. A valid matrix should have the block form "
+            "[W1 W2] where W1 @ W1.T.conj() + W2 @ W2.T.conj() = I and "
+            "W1 @ W2.T + W2 @ W1.T = 0."
+        )
+
+
+class FermionicGaussianState(QuantumCircuit):
+    """A circuit that prepares a fermionic Gaussian state."""
```

Comment:
I think you should add some documentation for this and the `SlaterDeterminant` class below, in terms of how these are supposed to be used.
You should also explicitly refer to the `QuadraticHamiltonian` which I assume these are meant to be worked with compared to other ansatze we have in Qiskit Nature like `UCC` which explicitly works with the `HartreeFock` initial state.